wagtail/images/views/serve.py

```diff
@@ -35,50 +59,68 @@ def as_view(cls, **initkwargs):
 
         return super().as_view(**initkwargs)
 
-    @method_decorator(cache_control(max_age=3600, public=True))
-    def get(self, request, signature, image_id, filter_spec, filename=None):
+    def get(
+        self,
+        request: "HttpRequest",
+        signature: str,
+        image_id: int,
+        filter_spec: str,
+        filename: Optional[str] = None,
+    ):
         if not verify_signature(
             signature.encode(), image_id, filter_spec, key=self.key
         ):
-            raise PermissionDenied
+            return self.get_error_response("Invalid signature", 400)
```

Comment:
drive-by suggestion: Change the codes to use `HTTPStatus`, so `HTTPStatus.BAD_REQUEST` instead of 400, and `HTTPStatus.GONE` instead of 410

Will try to review this properly soon.

Reply:
Gorgeous suggestion @zerolab! Implemented in [3e55491](https://github.com/wagtail/wagtail/pull/12833/commits/3e5549134d91dbf10c8b765a4dcad8579d288b6e)